### PR TITLE
Feature/implement forgot password

### PR DESCRIPTION
**Pull request by Liliia111**

### CODE REVIEW

File: SummerProject/config/settings.py [hunk 1]
```diff
@@ -25,6 +25,24 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
+EMAIL_BACKEND = "sendgrid_backend.SendgridBackend"
+
+EMAIL_HOST = 'smtp.sendgrid.net'
+
+EMAIL_HOST_USER = 'apikey'
+
+EMAIL_HOST_PASSWORD = SENDGRID_API_KEY
```
Comment (UlianaShevtsiv) [inline SummerProject/config/settings.py:34]: Could you please update the `readme` file with the instruction to add new `SENDGRID_API_KEY` variable local settings file? Also could you please share the `SENDGRID_API_KEY` variable value with the team-mates (if you hasn't done this yet)?

File: SummerProject/templates/email.html [hunk 2]
```diff
@@ -0,0 +1,344 @@
+<!doctype html>
```
Comment (UlianaShevtsiv) [inline SummerProject/templates/email.html]: Let's create a new directory `templates/emails/` and move this file there. Also lets name this file properly (eg. `forgot_password` or `reset_password`)?

File: SummerProject/user/urls.py [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
-from .views import registration, login, logout, UserView
+from .views import registration, login, logout, UserView, forgot_password_email_send, forgot_password_reset_confirm
 
 urlpatterns = [
     path('registration/', registration),
     path('login/', login),
     path('logout/', logout),
-    path('self/', UserView.as_view())
+    path('self/', UserView.as_view()),
+    path('forgotPassword/', forgot_password_email_send),
```
Comment (UlianaShevtsiv) [inline SummerProject/user/urls.py]: Lets follow the same pattern for the site URL names. Currently you use both camel-case (`forgotPassword/`) and snake-case (`reset_password_confirm/`). Please rename either `forgotPassword/` to use snake-case or `reset_password_confirm/` to use camel-case.

File: SummerProject/user/views.py [hunk 4]
```diff
@@ -70,3 +77,64 @@ def logout(request):
...
+        except (TypeError, ValueError, OverflowError, UserModel.DoesNotExist):
+            user = None
+
+        if user is not None and default_token_generator.check_token(user, token):
+            if is_valid_password_for_reset(data):
+                new_password = data['new_password_confirm']
+                user.set_password(new_password)
+                user.save()
+                return HttpResponse(status=201)
+            else:
```
Comment (UlianaShevtsiv) [inline SummerProject/user/views.py]: To simplify the code you can eliminate this `else` clause and the one below. As you have the `return HttpResponseBadRequest()` in line `140`

**Review by UlianaShevtsiv — APPROVED** [review]
+1 with couple of changes

**Review by AndriiStasiuk — APPROVED** [review]

**Review by filipenm — APPROVED** [review]

**Pull request merged**